src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -173,32 +192,50 @@ object CodeGenYul extends CodeGenerator {
                 new_name, // TODO rename transaction name (by adding prefix/suffix) iev: this seems to be done already
                 constructor.args.map(v => TypedName(v.varName, mapObsTypeToABI(v.typIn.toString))),
                 Seq(), //todo/iev: why is this always empty?
-                Block(constructor.body.flatMap((s: Statement) => translateStatement(s))))) //todo iev temp vars: likely a hack to work around something wrong
+                Block(constructor.body.flatMap((s: Statement) => translateStatement(s, None))))) //todo iev flatmap may be a bug to hide something wrong; None means that constructors don't return. is that true?
```

Comment:
Constructors probably will basically return addresses once we get addresses working.

Reply:
But this is probably okay for right now.

Reply:
ok!